### PR TITLE
[300] implements tests to MultipartBase64ImageMapper

### PR DESCRIPTION
**Pull request by ryhaldenys**

### CODE REVIEW

File: service/src/test/java/greencity/mapping/MultipartBase64ImageMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    @CsvSource({"mainFile,tempImage.jpg"})
+    public void convert_withCorrectData_isTrue(String fileName, String ordinalFileName){
+        File image = uploadImage();
+        String base64Image = encodeWithBase64(image);
+
+        var file = mapper.convert(base64Image);
+
+        assertThat(file.getName()).isEqualTo(fileName);
+        assertThat(file.getContentType()).isEqualTo(IMAGE_JPEG_VALUE);
+        assertThat(file.getOriginalFilename()).isEqualTo(ordinalFileName);
```
Comment (OlehTsipotan) [inline service/src/test/java/greencity/mapping/MultipartBase64ImageMapperTest.java:38]: `asserThat` -> `assertEquals`

File: service/src/test/java/greencity/mapping/MultipartBase64ImageMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    @CsvSource({"Incorrect image"})
+    public void convert_withIncorrectImageString_throwNullPointer(String image){
+        assertThatThrownBy(()->mapper.convert(image))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @SneakyThrows
+    @Test
+    public void convert_withThrownInMethodIOException_throwNotSavedException(){
+        File image = uploadImage();
```
Comment (OlehTsipotan) [inline service/src/test/java/greencity/mapping/MultipartBase64ImageMapperTest.java:60]: **TBD**

As for me, it is better to name this test:
`convert_whenIOExceptionThrown_expectNotSavedException` or ``convert_whenIOExceptionThrown_throwNotSavedException`

File: service/src/test/java/greencity/mapping/MultipartBase64ImageMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        String base64Image = encodeWithBase64(image);
+
+        var mockedImageIO = mockStatic(ImageIO.class);
+        mockedImageIO.when(()-> ImageIO.read((InputStream) any())).thenThrow(IOException.class);
+
+        assertThatThrownBy(()->mapper.convert(base64Image))
+                .isInstanceOf(NotSavedException.class)
+                .hasMessage("Cannot convert to BASE64 image");
+
+        mockedImageIO.close();
```
Comment (OlehTsipotan) [inline service/src/test/java/greencity/mapping/MultipartBase64ImageMapperTest.java:70]: **TBD**

try-with-resources block;

Not sure is this really worth refactoring. But try-with-resources are coll 4real.

```java
 try (MockedStatic<ImageIO> mockedImageIO = mockStatic(ImageIO.class)) {
        mockedImageIO.when(() -> ImageIO.read((InputStream) any())).thenThrow(IOException.class);

        assertThatThrownBy(() -> mapper.convert(base64Image))
                .isInstanceOf(NotSavedException.class)
                .hasMessage("Cannot convert to BASE64 image");
    }
```

**Review by OlehTsipotan — APPROVED** [review]
Good! Nothing that really has to be refactored.

**Review by Maryna-511750 — APPROVED** [review]

**Pull request merged**